odoo/models.py

```diff
@@ -1383,49 +1431,88 @@ def _extract_records(self, fields_, data, log=lambda a: None, limit=float('inf')
         * "id" is the External ID for the record
         * ".id" is the Database ID for the record
         """
-        fields = dict(self._fields)
-        # Fake fields to avoid special cases in extractor
-        fields = self._add_fake_fields(fields)
-        # m2o fields can't be on multiple lines so exclude them from the
-        # is_relational field rows filter, but special-case it later on to
-        # be handled with relational fields (as it can have subfields)
-        is_relational = lambda field: fields[field].relational
+        fields = self._fields
+
         get_o2m_values = itemgetter_tuple([
             index
-            for index, fnames in enumerate(fields_)
-            if fields[fnames[0]].type == 'one2many'
+            for index, fnames in enumerate(field_paths)
+            if fnames[0] in fields and fields[fnames[0]].type == 'one2many'
         ])
         get_nono2m_values = itemgetter_tuple([
             index
-            for index, fnames in enumerate(fields_)
-            if fields[fnames[0]].type != 'one2many'
+            for index, fnames in enumerate(field_paths)
+            if fnames[0] not in fields or fields[fnames[0]].type != 'one2many'
         ])
         # Checks if the provided row has any non-empty one2many fields
         def only_o2m_values(row):
             return any(get_o2m_values(row)) and not any(get_nono2m_values(row))
 
+        property_definitions = {}
+        property_columns = defaultdict(list)
+        for fname, *__ in field_paths:
+            if not fname:
+                continue
+            if '.' not in fname:
+                if fname not in fields:
+                    raise ValueError(f'Invalid field name {fname!r}')
+                continue
+
+            f_prop_name, property_name = fname.split('.')
+            if f_prop_name not in fields or fields[f_prop_name].type != 'properties':
+                # Can be .id
+                continue
+
+            definition = self.get_property_definition(fname)
+            if not definition:
+                # Can happen if someone remove the property, UserError ?
+                raise ValueError(f"Property {property_name!r} doesn't have any definition on {fname!r} field")
+
+            property_definitions[fname] = definition
+            property_columns[f_prop_name].append(fname)
+
+        # m2o fields can't be on multiple lines so exclude them from the
```

Comment:
It seems to me they are in the filter. Maybe you want to split with `is_multivalued`?

Reply:
Yep, the comment was not clear. m2o are not excluded here but not take in account in `only_o2m_values`, I suppose that it meant that. I am updating the comment.